src/freenet/pluginmanager/PluginManager.java

```diff
@@ -942,6 +952,27 @@ public FredPluginFCP getFCPPlugin(String plugname) {
 		return null;
 	}
 
+    /**
+     * Get the {@link FredPluginFCPMessageHandler.ServerSideFCPMessageHandler} of the plugin with
+     * the given class name.
+     * 
+     * @param pluginClassName
+     *            See {@link #getPluginInfoByClassName(String)}.
+     * @throws PluginNotFoundException
+     *             If the specified plugin is not loaded or does not provide an FCP server.
+     */
+    public FredPluginFCPMessageHandler.ServerSideFCPMessageHandler
+            getPluginFCPServer(String pluginClassName)
+                throws PluginNotFoundException{
+        
+        PluginInfoWrapper piw = getPluginInfoByClassName(pluginClassName);
+        if(piw != null && piw.isFCPServerPlugin()) {
+            return (FredPluginFCPMessageHandler.ServerSideFCPMessageHandler)piw.plug;
+        } else {
+            throw new PluginNotFoundException(pluginClassName);
```

Comment:
This function is used to create a FCP connection to a plugin. You aren't supposed to try to reconnect immediately if it fails, as the user is not expected to load/unload plugins frequently. Thus, this code can be assumed to be called infrequently, and thus the more object-oriented approach was chosen. 

ACK to keep as is?